### ISSUE TITLE
Recognize 1:1 color contrast as visually hidden rather than insufficient color contrast

### ISSUE THREAD

**Issue by nicklemmon** [dequelabs/axe-core#319]
I'm currently using a 'skip to main content' link that is visually hidden until it receives focus. Its container is white and the text is white (thus has 1:1 contrast) and is totally invisible. When it receives focus, the link receives a background color and the contrast ratio improves.

This works as expected, however, aXe indicates the element must have sufficient color contrast.

**Comment by marcysutton** [comment]
@dylanb @WilcoFiers how do you feel about checking the focus state of an element to see if the color changes? I feel uncomfortable exempting all 1-1 ratio elements without some additional heuristics. We'd have to focus each element to look at the focus pseudo-selector style, though. I could make it only do that if the ratio was 1-1.

For example, we have [tests for inputs](https://github.com/dequelabs/axe-core/blob/develop/test/integration/rules/color-contrast/color-contrast.html#L15) that fail if the colors match. Controls in this scenario are totally unusable. Shouldn't aXe warn you of that?

**Comment by WilcoFiers** [comment]
@marcysutton I have no problem of passing 1:1 ratio texts. As for focus style, the solution you propose doesn't work. Unfortunately, when you focus an element programatically, that doesn't change the element's style unless the OS focus is in the page itself, which it never is because we're calling axe from outside the page (command line, devtools panel, etc).

I do have a workaround for focus style which involves reading out the stylesheet itself, but that doesn't work cross domain. As far as I know, there is no consistent way to figure out what the focus style would be (and I've been trying to figure this one out for years, from way back when I worked on Quail).

**Comment by marcysutton** [comment]
Ok, I think we should put these items into Needs Review. That way if axe-core isn't catching something it should, like #273, it won't be suppressed completely (and we'd miss it).
